kafka-plugins-0.10/src/main/java/io/cdap/plugin/batch/source/KafkaBatchSource.java

```diff
@@ -167,10 +173,13 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     KafkaHelpers.setupKerberosLogin(kafkaConf, config.getPrincipal(), config.getKeytabLocation());
     kafkaConf.putAll(config.getKafkaProperties());
     kafkaRequests = KafkaInputFormat.saveKafkaRequests(conf, config.getTopic(), kafkaConf,
-                                                       config.getPartitions(), config.getMaxNumberRecords(),
+                                                       config.getPartitions(failureCollector),
```

Comment:
this should be a variable that was calculated at the start of the method.

Reply:
Before or after the call to getOrThrowException?

Reply:
Never mind, I just understood that you wanted me to move every reference to failureCollector to the beginning (including getPartitions), right before the call to getOrThrowException